app/models/census_employee.rb

```diff
@@ -559,11 +561,26 @@ def is_inactive?
     EMPLOYMENT_TERMINATED_STATES.include?(aasm_state)
   end
 
-  def active_or_pending_termination?
+  def is_cobra_possible?
+    return false if cobra_linked? || cobra_eligible? || rehired? || cobra_terminated?
     return true if self.employment_terminated_on.present?
     return true if PENDING_STATES.include?(self.aasm_state)
-    return false if self.rehired?
-    !(self.is_eligible? || self.employee_role_linked?)
+    
+    !(is_eligible? || employee_role_linked?)
+  end
+
+  def is_rehired_possible?
+    return false if cobra_linked? || cobra_eligible? || rehired?
+    return true if employment_terminated? || cobra_terminated?
+
+    !(is_eligible? || employee_role_linked?)
+  end
+
+  def is_terminate_possible?
```

Comment:
Can we replace this method with  may_terminate_employee_role? aasm_state machine helper ?